### ISSUE TITLE
config.php: handlaus deployssa

### ISSUE THREAD

**Issue by tverho** [Dominante/fermaatti#6]
deploy/post-receive sanoo nyt näin:
```
# 4. Use production config
`cd /var/wwwhome/jasensivut-kehitys/public/config; cp config.production.php config.php`
puts "DEPLOY: Using production config file (config.production.php)'
```
Ongelma 1: tuo tarvitsisi tomiakseen sudon jos/kun config.php ei ole käyttäjän omistuksessa
Ongelma 2: tuo rikkoo päivitykset jos se toimii, sillä `occ upgrade` vaatii että versionumero config.php:ssä vastaa edellisen päivityksen jälkeistä.

Mahdollinen ratkaisu olisi pitää minimaalinen config.php johon ei koskettaisi, ja tehdä erillinen `kehitys.config.php` jossa olisi kaikki muu. Owncloud huomioi kaikki .config.php -loppuiset tiedostot.

**Comment by k7hoven** [comment]
Joo, täällä onkin ehkä parempi jatkaa tätä. Eli tosta tulevaa virheilmoitusta ei pushin yhteydessä ehkä sitten näkynyt missään? Luulisin saavani yleispätevämmän ratkaisun (joskus aikanaan painiskelin hyvin samankaltaisen ongelman kanssa kun kehittelin versionhallintaa Fyysikkospeksin käsikirjoitustyökaluun XD). Toki ongelman vois myös kiertää jotenkin tolleen kun ehdotit, mutta hieman rajoittunutta on se.

**Comment by k7hoven** [comment]
Ei sinänsä, etteikö tota config-asiaa vois muutenkin vähän siistiä.

**Comment by tverho** [comment]
Pushin yhteydessä cp-komento valitti että oikeuksia ei ole. En välittänyt siitä suuremmin, koska koko komento on yleensä turha. Mutta sitten huomasin että tuo voi rikkoa päivityksen.

**Comment by tverho** [comment]
Periaatteessa se deploy-skripti voisi lukea sen version vanhasta configista ja lisätä uuteen, mutta tuntuu bugialttiilta.

**Comment by tverho** [comment]
Dokumentaation perusteella moni konfiguraatioparametri on autogeneroitu, eli niitä ei ole tarkoitus sörkkiä. Eli taitaa olla väärin kirjoittaa sen yli deployn yhteydessä. Sanoisin että kommentoidaan nyt tuo rivi post-receivessä, ja siirretään ehkä jatkossa osa configeista erilliseen tiedostoon joka on gitissä?
